File: capstone/Frontend/src/Services/QuestionService.js
import axios from 'axios';

const QUESTION_BASE_URL = 'http://localhost:8080/question';

class QuestionService{

    addQuestion(Question){
        return axios.post(QUESTION_BASE_URL, Question);
    }

    getAllQuestions(){
        return axios.get(QUESTION_BASE_URL);
    }

    getQuestionById(id){
        return axios.get(QUESTION_BASE_URL+'/'+id);
    }

    updateQuestion(id, Question){
        return axios.put(QUESTION_BASE_URL+'/'+id,Question);
    }

    deleteQuestion(id){
        return axios.delete(QUESTION_BASE_URL+'/'+id);
    } 
}

export default new QuestionService();

File: capstone/Backend/assesment-portal/src/main/java/com/project/assesmentportal/services/QuestionService.java
package com.project.assesmentportal.services;

import java.util.List;

import com.project.assesmentportal.dto.QuestionDto;

/**
 * Service interface for managing question-related operations.
 */
public interface QuestionService {
    /**
     * adds a new question.
     * @param questionDto of the question to be added.
     * @return QuestionDto
     */
    String addQuestion(QuestionDto questionDto);

    /**
     * gets all questions.
     * @return all the questions.
     */
    List<QuestionDto> getQuestions();

    /**
     * updates an existing question.
     * @param questionDto updates question.
     * @param questionId  of the question to be updated.
     * @return updated question.
     */
    String updateQuestion(QuestionDto questionDto, long questionId);

    /**
     * get question by question-id.
     * @param questionId of the question.
     * @return question.
     */
    QuestionDto getQuestionById(long questionId);

    /**
     * delete question by Id.
     * @param questionId id of the question to be deleted.
     */
    void deleteQuestion(long questionId);
}
